lib/scripts/graphics3d/volumeFilter3d.js

```diff
@@ -644,58 +654,295 @@ export default class VolumeFilter3d {
+        this.prevPos = targetX + targetY + targetZ;
+      }
+      console.log(`Target: ${targetX}, ${targetY}, ${targetZ}`);
+      this.lastMode.push(tools3dEraser.FILL);
+    } else {
+      targetX = x_;
+      targetY = y_;
+      targetZ = z_;
+    }
+    const intensityTarget = this.getIntensity(targetX, targetY, targetZ, undoFlag);
+    const stack = [];
+    stack.push({ 'tX':targetX, 'tY':targetY, 'tZ':targetZ });
+
+    if (!undoFlag) {
+      this.lastTarget.push(new THREE.Vector3(targetX, targetY, targetZ));
+    }
+
+    while (stack.length !== 0) {
+      console.log(`Size of stack: ${stack.length}`);
+      const point = stack.pop();
+      let openUp = false;
+      let openDown = false;
+      let openFar = false;
+      let openClose = false;
+      let xTmp = point.tX;
+      while (this.getIntensity(xTmp, point.tY, point.tZ, undoFlag) >= intensityTarget) {
+        xTmp--;
+      }
+      const leftBound = xTmp + 1;
+      xTmp = point.tX;
+      while (this.getIntensity(xTmp, point.tY, point.tZ, undoFlag) >= intensityTarget) {
+        xTmp++;
+      }
+      const rightBound = xTmp - 1;
+      for (xTmp = leftBound; xTmp <= rightBound; xTmp++) {
+        this.changeIntensity(xTmp, point.tY, point.tZ, undoFlag);
+        if (openUp === false) {
+          if (this.getIntensity(xTmp, point.tY + 1, point.tZ, undoFlag) >= intensityTarget) {
+            stack.push({ 'tX': xTmp, 'tY': (point.tY + 1), 'tZ': point.tZ });
+            openUp = true;
+          }
+        } else if (this.getIntensity(xTmp, point.tY + 1, point.tZ, undoFlag) < intensityTarget) {
+          openUp = false;
+        }
+
+        if (openDown === false) {
+          if (this.getIntensity(xTmp, point.tY - 1, point.tZ, undoFlag) >= intensityTarget) {
+            stack.push({ 'tX': xTmp, 'tY': (point.tY - 1), 'tZ': point.tZ });
+            openDown = true;
+          }
+        } else if (this.getIntensity(xTmp, point.tY - 1, point.tZ, undoFlag) < intensityTarget) {
+          openDown = false;
+        }
+
+        if (openFar === false) {
+          if (this.getIntensity(xTmp, point.tY, point.tZ + 1, undoFlag) >= intensityTarget) {
+            stack.push({ 'tX':xTmp, 'tY':point.tY, 'tZ':(point.tZ + 1) });
+            openFar = true;
+          }
+        } else if (this.getIntensity(xTmp, point.tY, point.tZ + 1, undoFlag) < intensityTarget) {
+          openFar = false;
+        }
+
+        if (openClose === false) {
+          if (this.getIntensity(xTmp, point.tY, point.tZ - 1, undoFlag) >= intensityTarget) {
+            stack.push({ 'tX':xTmp, 'tY':point.tY, 'tZ':(point.tZ - 1) });
+            openClose = true;
+          }
+        } else if (this.getIntensity(xTmp, point.tY, point.tZ - 1, undoFlag) < intensityTarget) {
+          openClose = false;
+        }
+      }
+    }
+    if (!undoFlag) {
+      this.updatableTextureMask.needsUpdate = true;
+    }
+  }
+  /*erasePixelsFloodFill(x_, y_, z_, startflag, mouseup, undoFlag) {
```

Comment:
Will we need this commented function in the future?

- [ ] Consider to remove commented function

Reply:
Just for time comparison with the main algorithm